torchvtk/augmentation/DictTransform.py

```diff
@@ -0,0 +1,155 @@
+import math
+import random
+from abc import abstractmethod
+from torchvtk.utils.volume_utils import make_5d
+import numpy as np
+import torch
+from torch.nn import functional as F
+
+
+class DictTransform(object):
+    """
+    Super Class for the Transforms.
+
+
+    """
+
+    def __init__(self, device="cpu", apply_on=["vol", "mask"], dtype=torch.float32, batch_transform=False):
+        """
+        :param device: The torch Code for the device on which the transformaion should be executed. Possiblities are ["cpu", "cuda"].
+        :param apply_on: The keys of the volumes that should be transformed.
+        :param dtype: The torch type in which the data are. Possibilities are [torch.float16, torch.float32].
+        """
+        super().__init__()
+
+        self.device = device
+        self.apply_on = apply_on
+        self.dtype = dtype
+        self.batch_transform = batch_transform
+
+    @abstractmethod
+    def transform(self, data):
+        pass
+
+    def __call__(self, data):
+
+        for key in self.apply_on:
+            tmp = data[key]
+
+            if not torch.is_tensor(tmp):
+                tmp = torch.from_numpy(tmp)
+
+            # to GPU.
+            if self.device == "cuda":
+                tmp = tmp.to(self.device)
+
+            if tmp.dtype is torch.float16:
+                tmp = tmp.to(torch.float32)
+
+            tmp = self.transform(tmp)
+
+            if self.dtype is torch.float16:
+                tmp = tmp.to(torch.float16)
+            elif self.dtype is not torch.float16 or torch.float32:
+                tmp = tmp.to(torch.float32)
+            tmp = tmp.to(self.device)
+            data[key] = tmp
+        return data
+
+
+class NoiseDictTransform(DictTransform):
```

Comment:
I just wonder should this have a clip parameter? Usually you would assume the data to be in a range, e.g. `[0,1]` and using a normal distribution can generate outliers from this. 
Should we enable this option or just implement a Clip Transform or so?